source/develop/release-management/features/network/networking-api-security-groups.html.md

```diff
@@ -0,0 +1,597 @@
+levels.
+
+The default behavior for the network's **port-security** attribute would be
+defined in the configuration file, through the ***port-security-enabled***,
+property, located in a new section, called **NETWORK**. This value will be set
+during *engine-setup* - refer to [this section](#engine-setup) - and the
+default choice will be 'yes', which will result in all traffic being dropped to
+the VMs.
+
+Existing ports can later be updated, disabling the **port-security** attribute,
+which would remove the ACLs that drop all IP traffic to the VM.
+
+As defined in the [network port security attribute definition](https://developer.openstack.org/api-ref/network/v2/#port-security), updating the **port-security** attribute on a network object **does not** cascade the value to ports attached to that network - meaning that effect would only apply to ports created *after* that update. Check the [upgrade](#installation/upgrade) section for the consequences of this requirement on ovirt-provider-ovn upgrades.
+
+## User workflow
+
+Since the feature is active by default, the intended workflow - from the oVirt
+engine user's perspective - is:
+
+* create an external network on the ovirt-provider-ovn - its default
+*port_security_enabled* value is true - which can be overriden during engine-setup.
+* create VMs attached to this external network - they will inherit the
+*port_security_attribute* from the network, which will result in dropping all
+IP traffic to that VM.
+* the intended traffic will have to be white-listed. Assuming the user is
+interested in allowing ssh traffic, the following ansible tasks should be
+used, which creates one security group - for which a security group rule
+allowing all **egress** traffic is automatically created - plus one security
+group rule allowing **ingress** tcp traffic meant for port 22:
+
+~~~~~~
+
+- os_security_group:
+    cloud: ovirt
+    state: present
+    name: my_app_default
+    description: allow ssh
+    register: my_app_default_security_group
+
+- os_security_group_rule:
+    cloud: ovirt
+    state: present
+    security_group: "{{ my_app_default_security_group.id }}"
+    direction: ingress
+    protocol: tcp
+    port_range_min: 22
+    port_range_max: 22
+
+~~~~~~
+
+* finally, the user is required to update the ports, indicating which security
+groups apply to it. The following ansible task shows how a single port, referenced
+by name, is updated:
+
+~~~~~~
+
+- os_port:
+    cloud: ovirt
+    state: present
+    name: <port_name>
+    security_groups:
+        - "{{ my_app_default_security_group.id }}"
+
+~~~~~~
+
+### Installation/Upgrade
+
+The install time required updates are described in the section [below](#engine-setup).
+
+On upgrades with existing VMs - having attachments to external networks -
+the ports will **not** be updated, and it will be up to the administrator to
+manually activate port security in the existent ports.
+
```

Comment:
Very valid point; I think that since at the moment neither [os_network](https://docs.ansible.com/ansible/2.5/modules/os_network_module.html) nor [os_port](https://docs.ansible.com/ansible/2.6/modules/os_port_module.html) ansible modules allow updating the **port_security_enabled** attribute, we should provide an upgrade script that uses the ovn-nb CLI to set those values (port_security at network level) to false.

sounds  like a feasible option ?

Reply:
Why not just let the ovirt-provider-ovn interpret 'undefined' for networks in the db as port_security_enabled=`false` ?

Reply:
Even nicer.

To sum it up, when new networks are created - and port security is not provider - I'll set the value from the configuration file (created from the user answers to engine-config); for existing networks, a default of False will be assumed.

This will lead to OVN networks created without using the ovirt-provider-ovn using this 'False' default, meaning no security groups will apply to their ports.

Implementing it is a matter of simply changing [this line](https://gerrit.ovirt.org/#/c/93935/11/provider/neutron/neutron_api.py@1158) back to the 'False' default.

Do you foresee any issue with that approach ?

Reply:
Sounds good.